cmake/CMakeLists.txt

```diff
@@ -109,6 +109,7 @@ option(onnxruntime_PREFER_SYSTEM_LIB "Experimental: Build with the preinstalled
 option(onnxruntime_ENABLE_NVTX_PROFILE "Enable NVTX profile." OFF)
 option(onnxruntime_ENABLE_TRAINING "Enable training functionality." OFF)
 option(onnxruntime_ENABLE_TRAINING_E2E_TESTS "Enable training end-to-end tests." OFF)
+option(onnxruntime_USE_MPI "Build with MPI support, required by HOROVOD and NCCL" ON)
```

Comment:
tools/ci_build/build.py has been used in the team to build ORT. How about updating build.py to enable/disable it?

Reply:
Sorry, I wasn't very clear here -- I was thinking of this PR as an initial change to get a distinct choice for builds using MPI visible during compilation, and then having some further discussion next week on how to expose via build.py.

Currently for build.py we have a mix of boolean flags like --use_horovod, and flags that provide configuration settings like --mpi_home (without a separate --use_* flag).  I think it would be cleaner for us to have both of these flags for each of the dependencies, as we do already for CUDA.  Having --use_* explicit lets us fail early in builds if the user requests a feature that is not found.  Having --*_home distinct lets the user decide whether or not to override the find_package logic in cmake.

Reply:
Agree. I thought to have --use_mpi for MPI initially but then changed the idea. I was not sure whether introducing --use_mpi would cause more confusion since --use_nccl or --use_horovod would request mpi to be enabled by default. For example, if --use_nccl is specified in the command option, will the user still need to have --use_mpi in the command option?

Reply:
+1 here.  NCCL requires MPI to do the initialization.  We should either:
[1] When using NCCL, automatically set `--use_mpi` and display a log message to build OR
[2] When using NCCL but not `--use_mpi`, display an error and fail the build.